### DISCUSSION TITLE
Load a file as byte array?

### DISCUSSION THREAD

**Discussion by laurentsenta** [vercel/next.js#11886] [Help]
Hi there, thanks for maintaining this framework, it's a pleasure to use!

I'm lost on file loading:

I'm trying to write an API endpoint that loads a pdf file as binary data, process it, and sends it to the user.

I would like to load my template data as a byte buffer, something like:

```
import pdfBase from './template.pdf'
const existingPdfBytes = pdfBase.arrayBuffer
``` 

I had an error,

```
Module parse failed: Unexpected token (1:0)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)
```

I tried to add `file-loader` to my `next.config.js`:

```
    webpack: (config, options) => {
      config.module.rules.push({
        test: /\.pdf/,
        loader: 'file-loader',
      })

      return config
```

but it looks like all I get is a name / URL,

is there a recommended way to load data in the API?

**Comment by Timer** [comment]
If this is an API endpoint, it'd be best to use `fs` to read the file.

**Reply by laurentsenta** [reply]
I see @Timer thanks for the info, is there any doc on how I should / could get the file path?

If I understand correctly, I'm stuck until your feature request is resolved: https://github.com/zeit/next.js/issues/8251

(I resorted to encoding my file as a base64 string, stored in a JSON I import directly thanks to Typescript, hackish).

**Reply by Timer** [reply]
You can read from `process.cwd()` and build your path that way.
